PathKit/PathKit.swift

```diff
@@ -90,94 +108,557 @@ public struct Path : Equatable, Hashable, CustomStringConvertible, StringLiteral
+        return data.writeToFile(normalize().path, atomically: true)
     }
 
-    public func write(string:String) -> Bool {
-        if let data = string.dataUsingEncoding(NSUTF8StringEncoding, allowLossyConversion: true) {
+    /// Reads the file.
+    ///
+    /// - Note: Works atomically: the data is written to a backup file, and then — assuming no
+    ///   errors occur — the backup file is renamed to the name specified by path.
+    ///
+    /// - Parameter string: the string to write to file.
+    ///
+    /// - Parameter encoding: the encoding which should be used to represent the string as bytes.
+    ///   (by default: `NSUTF8StringEncoding`)
+    ///
+    /// - Returns: the contents of the file at the specified path as string.
+    ///
+    public func write(string: String, encoding: NSStringEncoding = NSUTF8StringEncoding) -> Bool {
+        if let data = string.dataUsingEncoding(encoding, allowLossyConversion: true) {
             return write(data)
         }
 
         return false
     }
+}
+
 
-    // MARK: Children
+// MARK: Traversing
 
-    public func children(directories directories:Bool = true) throws -> [Path] {
-        let manager = NSFileManager()
-        let contents = try manager.contentsOfDirectoryAtPath(path)
-        let paths = contents.map {
+extension Path {
+    /// Get the parent directory
+    ///
+    /// - Returns: the normalized path of the parent directory
+    ///
+    public func parent() -> Path {
+        return self + ".."
+    }
+    
+    /// Performs a shallow enumeration in a directory
+    ///
+    /// - Returns: paths to all files, directories and symbolic links contained in the directory
+    ///
+    public func children() throws -> [Path] {
+        return try Path.fileManager.contentsOfDirectoryAtPath(path).map {
+            self + Path($0)
+        }
+    }
+    
+    /// Performs a deep enumeration in a directory
+    ///
+    /// - Returns: paths to all files, directories and symbolic links contained in the directory or
+    ///   any subdirectory.
+    ///
+    public func recursiveChildren() throws -> [Path] {
+        return try Path.fileManager.subpathsOfDirectoryAtPath(path).map {
             self + Path($0)
         }
+    }
+}
+
+
+// MARK: SequenceType
+
+extension Path : SequenceType {
+    /// Enumerates the contents of a directory, returning the paths of all files and directories
+    /// contained within that directory. These paths are relative to the directory.
+    public struct DirectoryEnumerator : GeneratorType {
+        public typealias Element = Path
 
-        if directories {
-            return paths
+        let path: Path
+        let directoryEnumerator: NSDirectoryEnumerator
+
+        init(path: Path) {
+            self.path = path
+            self.directoryEnumerator = Path.fileManager.enumeratorAtPath(path.path)!
+        }
+
+        public func next() -> Path? {
+            if let next = directoryEnumerator.nextObject() as! String? {
```

Comment:
Nope, that's something different. This can't be better denoted in Foundation because `NSEnumerator` isn't generic, so a force-cast is the only way. If this is not a NSString reference which can be bridged to an optional string than something is really wrong in Foundation and this shouldn't be silently ignored.